dict_inventory.py

```diff
@@ -44,7 +46,17 @@ def delete_product(self, product_id):
       raise KeyError('No product with product id %d' % product_id)
 
   def test_init(self):
-    self.products[1] = Product({PRODUCT_ID: 1, LOCATION_ID: 1,
+    self.products[0] = Product({PRODUCT_ID: self.get_next_product_id(),
+                                LOCATION_ID: self.get_next_location_id(),
                                 USED: 1,NEW: 1, OPEN_BOX: 1, RESTOCK_LEVEL: 11})
-    self.products[2] = Product({PRODUCT_ID: 2, LOCATION_ID: 2,
-                                USED: 2,NEW: 2, OPEN_BOX: 2, RESTOCK_LEVEL: 22})
\ No newline at end of file
+    self.products[1] = Product({PRODUCT_ID: self.get_next_product_id(),
+                                LOCATION_ID: self.get_next_location_id(),
+                                USED: 2,NEW: 2, OPEN_BOX: 2, RESTOCK_LEVEL: 22})
+
+  def get_next_product_id(self):
+    self.next_product_id += 1
+    return self.next_product_id - 1
+
+  def get_next_location_id(self):
+    self.next_location_id += 1
```

Comment:
Purely increasing location_id may have empty gap between product location. Like if position 3's product is deleted, next product just created in the last location id+1, like 100+1, but 3 is empty and no product is going to be stored in it again.